Code/Framework/AzFramework/AzFramework/Input/Buses/Requests/InputDeviceRequestBus.h

```diff
@@ -227,7 +227,7 @@ namespace AzFramework
         //! Passing InputDeviceType::Implementation::Create as the argument will create the default
         //! device implementation, while passing nullptr will delete any existing implementation.
         //! \param[in] implementationFactory Pointer to the function that creates the implementation.
-        virtual void SetCustomImplementation(typename InputDeviceType::ImplementationFactory implementationFactory) = 0;
+        virtual void SetCustomImplementation(typename InputDeviceType::ImplementationFactory* implementationFactory) = 0;
```

Comment:
How about have this function accept a non-const lvalue reference to the ImplementationFactory?

That way the nullptr check is put on the caller.
```suggestion
        virtual void SetCustomImplementation(typename InputDeviceType::ImplementationFactory& implementationFactory) = 0;
```